### PR TITLE
Transfer Anchor component props

### PR DESCRIPTION
**Pull request by epilande**
#### What does this PR do?

Transfer Anchor component props to underlying DOM element.

#### What testing has been done on this PR?

Tested on grommet docs page.

#### How should this be manually tested?

Add props to `Anchor`.

#### What are the relevant issues?

https://github.com/grommet/grommet/issues/85

#### Is this change backwards compatible or is it a breaking change?

Yes, backwards compatible.

### CODE REVIEW

File: src/js/components/Anchor.js [hunk 1]
```diff
@@ -41,32 +46,27 @@ export default class Anchor extends Component {
...
 
-    const Component = this.props.tag;
+    const Component = tag;
     return (
-      <Component id={this.props.id} className={classes}
-        href={this.props.href}
-        target={this.props.target}
-        onClick={this.props.onClick}
-        aria-label={this.props.a11yTitle}>
+      <Component {...props} id={id} className={classes} aria-label={a11yTitle}>
```
Comment (epilande) [inline src/js/components/Anchor.js]: We can't remove `id={id}`; since we're assigning `id` on line 15, `{...props}` doesn't contain `id`.

Comment (epilande) [inline src/js/components/Anchor.js]: Removed `aria-label`

Comment (epilande) [inline src/js/components/Anchor.js]: Or I can remove the assignment of `id` from line 15 & reference it on line 25 as `props.id`

File: src/js/components/Anchor.js [hunk 2]
```diff
@@ -41,32 +46,27 @@ export default class Anchor extends Component {
 
     let classes = classnames(
       CLASS_ROOT,
-      this.props.className,
+      className,
```
Comment (ericsoderberghp) [inline src/js/components/Anchor.js]: Let's put any provided className prop last in the list. It makes the class easier to read when inspecting in the browser.

**Comment by alansouzati** [comment]
hey manny, do you mind fixing the conflicts here? sorry.

**Comment by epilande** [comment]
@alansouzati Fixed the merge conflicts

**Pull request merged**